### DISCUSSION TITLE
Can't install on windows using npx-create-directus-project

### DISCUSSION THREAD

**Discussion by jaylinsell** [directus/directus#7725] [Archive]
### Preflight Checklist

- [X] I have completed all [Troubleshooting Steps](https://docs.directus.io/getting-started/support/#troubleshooting-steps).
- [X] I'm on [the latest version of Directus](https://github.com/directus/directus/releases).
- [X] There's [no other issue](https://github.com/directus/directus/issues) that already describes my problem.

### Describe the Bug

When trying to install Directus for the first time, the install fails due to what seems to be an argon package.

I've attached the end of the logs where the build fails:

> 12686 verbose stack Error: argon2@0.28.2 install: `node-pre-gyp install --fallback-to-build`
> 12686 verbose stack Exit status 1
> 12686 verbose stack     at EventEmitter.<anonymous> (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
> 12686 verbose stack     at EventEmitter.emit (events.js:376:20)
> 12686 verbose stack     at ChildProcess.<anonymous> (C:\Program Files (x86)\nodejs\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
> 12686 verbose stack     at ChildProcess.emit (events.js:376:20)
> 12686 verbose stack     at maybeClose (internal/child_process.js:1055:16)
> 12686 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
> 12687 verbose pkgid argon2@0.28.2
> 12688 verbose cwd D:\Clients\Dev Sites\directusdemo
> 12689 verbose Windows_NT 10.0.19042
> 12690 verbose argv "C:\\Program Files (x86)\\nodejs\\node.exe" "C:\\Program Files (x86)\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "directus" "--production" "--no-optional"
> 12691 verbose node v14.17.0
> 12692 verbose npm  v6.14.13
> 12693 error code ELIFECYCLE
> 12694 error errno 1
> 12695 error argon2@0.28.2 install: `node-pre-gyp install --fallback-to-build`
> 12695 error Exit status 1
> 12696 error Failed at the argon2@0.28.2 install script.
> 12696 error This is probably not a problem with npm. There is likely additional logging output above.
> 12697 verbose exit [ 1, true ]

I'm on Windows, using Node 4.17.0.

Have tried the solutions for a similar issue logged where the install wouldn't work on Centos - but have had no luck.

### To Reproduce

Run `npx create-directus-project` on windows.

### What version of Directus are you using?

Latest

### What version of Node.js are you using?

14.17.0

### What database are you using?

Haven't got this far yet

### What browser are you using?

Haven't got this far yet

### What operating system are you using?

Windows 10

### How are you deploying Directus?

Haven't got here yet.

**Comment by nickrum** [comment]
It looks like you are running the 32 Bit version of Node. The `argon2` package only provides binaries for 64 Bit systems, which means it will try to build those binaries for your platform. This requires [additional tools](https://github.com/nodejs/node-gyp#on-windows) to be installed. If your system actually supports 64 Bits, you could try installing the 64 Bit version of Node.

**Comment by rijkvanzanten** [comment]
To run Directus, you currently need Node 12.20 or newer. 

BTW, Node 4 has been end-of-life for years now, so I would highly recommend updating to a more recent version (16 is the current)

**Comment by jaylinsell** [comment]
My bad, that was a typo - I was on Node V **14.17.0**, not 4.17.0.

I've reinstalled Node for 64bit, have also re-installed all the windows build tools as described here for windows: https://github.com/nodejs/node-gyp.
I've now been able to 'install' directus, so to speak, but I'm now having issues choosing a database from the node prompt.

Steps:
1. Run `npx create-directus-project demo-proj`
2. Select Postgres for the database

This is now where the install fails.

> ? Choose your database client PostgreSQL / Redshift
> ⠙ Installing Database Driver...Error: Command failed with ENOENT: npm install pg --production
> spawn cmd.exe ENOENT
>     at Process.ChildProcess._handle.onexit (internal/child_process.js:269:19)
>     at onErrorNT (internal/child_process.js:467:16)
>     at processTicksAndRejections (internal/process/task_queues.js:82:21) {
>   errno: -4058,
>   code: 'ENOENT',
>   syscall: 'spawn cmd.exe',
>   path: 'cmd.exe',
>   spawnargs: [
>     '/q',
>     '/d',
>     '/s',
>     '/c',
>     '"npm ^"install^" ^"pg^" ^"--production^""'
>   ],
>   originalMessage: 'spawn cmd.exe ENOENT',
>   shortMessage: 'Command failed with ENOENT: npm install pg --production\n' +
>     'spawn cmd.exe ENOENT',
>   command: 'npm install pg --production',
>   escapedCommand: 'npm install pg --production',
>   exitCode: undefined,
>   signal: undefined,
>   signalDescription: undefined,
>   stdout: '',
>   stderr: '',
>   failed: true,
>   timedOut: false,
>   isCanceled: false,
>   killed: false
> }
> ✖ Installing Directus
> Error: undefined

**Comment by rijkvanzanten** [comment]
From a quick google, that messages `span cmd.exe ENOENT` seems to indicate that `cmd` isn't available in your `PATH` under WSL. I'm not a windows user myself, but it seems like this is most likely related to your specific way of running Node on your machine, and not an issue we can solve in the Directus tool necessarily. I'll transfer this to a Q&A discussion 👍🏻

**Comment by rijkvanzanten** [comment]
Here's some references that might point you in the right direction: https://github.com/facebook/create-react-app/issues/7251, https://github.com/facebook/create-react-app/issues/7094#issuecomment-494611206, https://stackoverflow.com/questions/28624686/get-spawn-cmd-enoent-when-try-to-build-cordova-application-event-js85/28626189#28626189

**Reply by jaylinsell** [reply]
Thanks Rijk.

Yep, turns out it was my setup.
If you need any additional info on what I had to do for documentation / pre-requirements for Windows users, here's the summary:

Windows Build Tools also needs to be installed with Node.
If it's not, you can install it from here: https://visualstudio.microsoft.com/thank-you-downloading-visual-studio/?sku=BuildTools
You then need to run: **npm config set msvs_version 2019** (or what ever the version year is)

Then check if **C:\Windows\System32** is added to the PATH list of systems environment variables.
 
 Then it all runs a treat.
There's a small few issues I've hit with certain fields creating 500 errors, but I assume that's due to my mis-configuration of the fields. Only concern is it doesn't tell me what I've done wrong. 
 
Awesome work by the way, it's really nice to use, and has some nice features Strapi hasn't been able to include.
Really looking forward to using this in my projects.
